contracts/extensions/BasicIssuanceExtension.sol

```diff
@@ -0,0 +1,248 @@
+     * owner and methodologist, and sends to owner fee recipient and methodologist respectively.
+     */
+    function distributeFees(ISetToken _setToken) public {
+        IDelegatedManager delegatedManager = _manager(_setToken);
+
+        uint256 totalFees = _setToken.balanceOf(address(delegatedManager));
+
+        address methodologist = delegatedManager.methodologist();
+        address ownerFeeRecipient = delegatedManager.ownerFeeRecipient();
+
+        uint256 ownerTake = totalFees.preciseMul(delegatedManager.ownerFeeSplit());
+        uint256 methodologistTake = totalFees.sub(ownerTake);
+
+        if (ownerTake > 0) {
+            delegatedManager.transferTokens(address(_setToken), ownerFeeRecipient, ownerTake);
+        }
+
+        if (methodologistTake > 0) {
+            delegatedManager.transferTokens(address(_setToken), methodologist, methodologistTake);
+        }
+
+        emit FeesDistributed(address(_setToken), ownerFeeRecipient, methodologist, ownerTake, methodologistTake);
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external {
+        ISetToken setToken = _delegatedManager.setToken();
+
+        require(
+            managerCore.isFactory(msg.sender) || 
+            address(_delegatedManager) == setToken.manager(),
+            "Must be factory or input must be SetToken manager"
+        );
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        setManagers[setToken] = _delegatedManager;
+
+        _delegatedManager.initializeExtension();
+
+        ExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager and BasicIssuanceModule to the SetToken
+     *
+     * @param _delegatedManager             Instance of the DelegatedManager to initialize
+     * @param _maxManagerFee                Maximum fee that can be charged on issue and redeem
+     * @param _managerIssueFee              Fee to charge on issuance
+     * @param _managerRedeemFee             Fee to charge on redemption
+     * @param _feeRecipient                 Address to send fees to
+     * @param _managerIssuanceHook          Instance of the Manager Contract with the Pre-Issuance Hook function
+     */
+    function initializeModuleAndExtension(
+        IDelegatedManager _delegatedManager,
+        uint256 _maxManagerFee,
+        uint256 _managerIssueFee,
+        uint256 _managerRedeemFee,
+        address _feeRecipient,
+        address _managerIssuanceHook
+    ) 
+        external 
+    {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
```

Comment:
I think here we should also require that the factory is NOT the sender if our intention is that this is only called after the manager initialization is done. We should also add the `isValidManager` modifier here as well